channels/OFF-CHAIN.md

```diff
@@ -0,0 +1,503 @@
+- `channel_reserve`: the minimum amount both peers need to maintain. This makes
+  sure that both have to lose something in case they act maliciously
+- `responder_pubkey`: the account that the initiator wants to use to open the
+  channel
+
+(***TODO***: This could be interactive, i.e. if the responding peer sends
+different amounts, then that might communicate that it wants these instead.)
+
+
+#### Requirements
+
+*Responder*:
+
+- `chain_hash` MUST identify the chain to be used
+- `temporary_channel_id` MUST be unique between the involved peers
+- `lock_period` SHOULD be sufficient time to safely publish transactions to the
+  blockchain to stop a cheater
+- `responder_pubkey` MUST be a valid ed25519 pubkey
+
+
+### `funding_created`
+
+In order to open a channel on chain, peers need to cooperate and co-sign the
+`channel_create` transaction. The `funding_created` message is used by the
+initiator to send their signature to the responder.
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| temporary_channel_id | 32 |
+ ---------------------- ----
+| transaction_data     | 32 |
+ ---------------------- ----
+| signature            | 64 |
+ ---------------------- ----
+```
+
+
+#### Requirements
+
+*Responder*:
+
+- MUST abort if the signature is invalid for the provided transaction data
+
+
+### `funding_signed`
+
+If the responder was able to validate the initiator's signature sent in the
+`funding_created` message, then it should proceed to respond with its
+`funding_signed` message, unless the initiator specified a `push_amount > 0` in
+the `channel_open` message. In the latter case, they should wait for the
+initiator to send the initial state signed by them.
+
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| temporary_channel_id | 32 |
+ ---------------------- ----
+| signature            | 64 |
+ ---------------------- ----
+```
+
+
+### `funding_locked`
+
+Opening a channel requires an on-chain transaction. This transaction needs to be
+included in a block and, since we only have probabilistic finality, be
+sufficiently confirmed, s.t. the probability of a chain re-organisation is
+negligible.
+
+This message is exchanged by both peers to signal to each other that the above
+condition has been met from their point of view and only after both of them
+agree on this, can the channel be considered to be opened.
+
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| temporary_channel_id | 32 |
+ ---------------------- ----
+| transaction_id       | 32 |
+ ---------------------- ----
+```
+
+
+#### Requirements
+
+A node MUST NOT send the `funding_locked` message unless the `channel_create`
+transaction has `minimum_depth` confirmations.
+
+## Initialise contract
+
+### `update_init_contract`
+
+
+## State update
+
+State updates require consent of both peers.
+
+Each update MUST have a strictly increasing sequence number, which SHOULD start
```

Comment:
I understand there is only one sequence number for all state updates - i.e. not one sequence number for participant. Is there a suggested approach for dealing with participants sending to each other update with same sequence number at the same time?